app/services/post_owner_changer.rb

```diff
@@ -13,7 +13,7 @@ def initialize(params)
   def change_owner!
     ActiveRecord::Base.transaction do
       @post_ids.each do |post_id|
-        post = Post.with_deleted.where(id: post_id, topic_id: @topic.id).first
+        post = Post.with_deleted.find_by(id: post_id, topic_id: @topic.id)
```

Comment:
This, albeit correct is not inline with our guidlines per:

https://meta.discourse.org/t/discourse-development-contribution-guidelines/3823

> "Refactoring only NOT welcome"